src/nix/profile.cc

```diff
@@ -329,7 +329,63 @@ struct CmdProfileInstall : InstallablesCommand, MixDefaultProfile
             manifest.elements.push_back(std::move(element));
         }
 
-        updateProfile(manifest.build(store));
+        try {
+            updateProfile(manifest.build(store));
+        } catch (BuildProfileConflictError & conflictError) {
+            std::string fallbackInstallable = "INSTALLABLE";
+            std::string conflictingInstallable = [&]() {
+                auto maybeConflictingStorePath = store->maybeParseStorePath(conflictError.conflictingPkgPath);
+                if (!maybeConflictingStorePath) {
+                    printError("warning: unable to determine installable from conflicting package path (%1%)", conflictError.conflictingPkgPath);
+                    return fallbackInstallable;
+                }
+                auto conflictingStorePath = *maybeConflictingStorePath;
+                auto conflictingManifestElement = find_if(manifest.elements.begin(), manifest.elements.end(), [&](const ProfileElement & currentProfileElement) {
+                    return currentProfileElement.storePaths.contains(conflictingStorePath);
+                });
+                if (conflictingManifestElement == manifest.elements.end()) {
+                    printError("warning: unable to find profile element for conflicting store path (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                if (!conflictingManifestElement->source) {
+                    printError("warning: conflicting store path does not have a source (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                return conflictingManifestElement->source->originalRef.to_string();
+            }();
+
+            throw Error(
+                "An existing package already provides the following file to the user's profile:\n"
+                "\n"
```

Comment:
Probably nicer to write this using multiline string literals, i.e. `R"(...)"`.

Reply:
I have looked into various ways of doing multiline strings in C++. I think R"(...)" was one of them and it seemed it didn't support removing indentation, so it required me to put all text on the first column. The current solution seemed like one that did properly indent, which looked a bit better.